Armadillo/src/main/java/com/scribd/armadillo/playback/ExoPlaybackExceptionExt.kt

```diff
@@ -47,4 +65,17 @@ internal fun ExoPlaybackException.toArmadilloException(): ArmadilloException {
     } else {
         UnexpectedException(cause = this, actionThatFailedMessage = "Exoplayer error")
     }
+}
+
+private fun DataSpec.toAnalyticsMap(): Map<String, String> {
+    return mapOf(
+        "uri" to uri.toString(),
+        "uriPositionOffset" to uriPositionOffset.toString(),
+        "httpMethod" to httpMethod.toString(),
+        "position" to position.toString(),
+        "length" to length.toString(),
+        "key" to key.toString(),
+        "flags" to flags.toString(),
+        "customData" to customData.toString()
+    )
```

Comment:
One thing i didn't have the time to build yet was internet status detection.  I've pulled off the more obvious connectivity errors separate from the HTTP status ones, but I believe there's still internet connectivity issues still being mixed into here.

Reply:
I'll add this in the everand integration itself, since we have a NetUtils method for checking internet status, and other logs are also using it this way.